src/Test/Perf/ConsumptionParser/JsonResultParser.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using Newtonsoft.Json.Linq;
+
+namespace Roslyn.Test.Performance.Utilities.ConsumptionParser
+{
+
+    internal class RunInfo 
+    {
+        public string Username{ get; }
+        public string Branch { get; }
+
+        public RunInfo(string username, string branch)
+        {
+            this.Username = username;
+            this.Branch = branch;
+        }
+
+        public static RunInfo Parse(String s)
+        {
+            dynamic obj = JObject.Parse(s);
+            String username = obj.roots[0].job.user.userName;
```

Comment:
(Also, I'm not a big fan of dynamic with json.net, can we be more strongly typed if we're creating the files?)